libraries/APM_Control/AR_PosControl.cpp

```diff
@@ -192,19 +194,21 @@ void AR_PosControl::update(float dt)
     const Vector2f vel_target_FR = AP::ahrs().earth_to_body2D(_vel_target);
 
     // desired speed is normally the forward component (only) of the target velocity
-    // but we do not let it fall below the minimum turn speed unless the vehicle is slowing down
-    const float abs_des_speed_min = MIN(_vel_target.length(), turn_speed_min);
-    float des_speed;
-    if (_reversed != backing_up) {
-        // if reversed or backing up desired speed will be negative
-        des_speed = MIN(-abs_des_speed_min, vel_target_FR.x);
-    } else {
-        des_speed = MAX(abs_des_speed_min, vel_target_FR.x);
+    float des_speed = vel_target_FR.x;
+    if (!stopping) {
+        // do not let target speed fall below the minimum turn speed unless the vehicle is slowing down
+        const float abs_des_speed_min = MIN(_vel_target.length(), turn_speed_min);
```

Comment:
The `stopping` bool is set from the desired velocity `_vel_desired`. 

Here we do the min of `_vel_target`, so if the target and desired were both zero then `abs_des_speed_min` should also be 0. 

I think this discrepancy might cause odd stuff if you were to just input a position target where `_vel_desired` would always be zero?